gradle-plugin/src/main/groovy/org/groovymc/modsdotgroovy/gradle/MDGExtension.groovy

```diff
@@ -2,135 +2,141 @@ package org.groovymc.modsdotgroovy.gradle
 
-    MapProperty<Platform, String[]> getPlatforms() {
-        return platforms
-    }
+        this.modsDotGroovyFile.convention(sourceSet.resources.matching {
+            include DEFAULT_MDG
+        })
 
-    SetProperty<String> getEnvironmentBlacklist() {
-        return environmentBlacklist
-    }
+        this.multiplatform = project.objects.property(Boolean)
+        this.multiplatform.convention(false)
 
-    Property<Boolean> getSetupDsl() {
-        return setupDsl
+        this.setupPlugins.convention(false)
+        this.setupTasks.convention(false)
+        this.setupDsl.convention(false)
     }
 
-    Property<Boolean> getSetupPlugins() {
-        return setupPlugins
+    void conversionOptions(Action<MDGConversionOptions> action) {
+        action.execute(conversionOptions)
     }
 
-    Property<Boolean> getSetupTasks() {
-        return setupTasks
+    private static List<Platform> inferPlatforms(Project project) {
+        if (project.plugins.findPlugin('net.minecraftforge.gradle')) return List.of(Platform.FORGE)
+        else if (project.plugins.findPlugin('net.neoforged.gradle.userdev')) return List.of(Platform.NEOFORGE)
+        else if (project.plugins.findPlugin('fabric-loom')) return List.of(Platform.FABRIC)
+        else if (project.plugins.findPlugin('org.quiltmc.loom')) return List.of(Platform.QUILT)
+        else return List.of()
     }
 
-    void setPlatforms(Map<Platform, ?> platforms) {
-        this.platforms.set(platforms.collectEntries { Platform platform, def paths ->
-            if (paths instanceof String) return [platform, [paths] as String[]]
-            else if (paths instanceof List<String>) return [platform, paths.toArray(String[]::new)]
-            else throw new IllegalArgumentException("Invalid platform paths: $paths. Expected String or List<String>, got ${paths.getClass().getName()}")
-        })
+    void multiplatform(Action<Multiplatform> action) {
+        action.execute(new Multiplatform())
     }
 
-//    void setPlatform(Platform platform) {
-//        final String[] projectPath = [project.path]
-//        this.platforms.set(Map.of(platform, projectPath))
-//    }
-
-    void setEnvironmentBlacklist(String... blacklist) {
-        this.environmentBlacklist.set(Set.of(blacklist))
+    void expose(Object file, Action<? super ConfigurablePublishArtifact> configureAction) {
+        setupDsl.set(false)
+        setupPlugins.set(false)
+        setupTasks.set(false)
+        var configurationName = forSourceSetName(sourceSet.name, EXPOSE_SOURCE_SET)
+        var exposingConfiguration = project.configurations.maybeCreate(configurationName)
+        exposingConfiguration.canBeResolved = false
+        exposingConfiguration.canBeConsumed = true
+
+        project.artifacts {
+            add(configurationName, file, configureAction)
+        }
     }
 
-    void setSetupDsl(boolean setupDsl) {
-        this.setupDsl.set(setupDsl)
+    void expose(Object file) {
+        expose(file, {})
```

Comment:
Out of curiosity, does the compiled output differ if you do `() -> {}` instead?

Reply:
That would be invalid - an action expects an input